Helper/Entity/ProductHelper.php

```diff
@@ -20,19 +21,22 @@
 use Magento\Catalog\Model\Product\Type\AbstractType;
 use Magento\Catalog\Model\Product\Visibility;
 use Magento\Catalog\Model\ResourceModel\Eav\Attribute as AttributeResource;
+use Magento\Catalog\Model\ResourceModel\Product\CollectionFactory;
 use Magento\CatalogInventory\Api\StockRegistryInterface;
 use Magento\CatalogInventory\Helper\Stock;
 use Magento\ConfigurableProduct\Model\Product\Type\Configurable;
+use Magento\Customer\Model\ResourceModel\Group\Collection as GroupCollection;
 use Magento\Directory\Model\Currency as CurrencyHelper;
 use Magento\Eav\Model\Config;
 use Magento\Framework\DataObject;
 use Magento\Framework\Event\ManagerInterface;
-use Magento\Framework\ObjectManagerInterface;
 use Magento\Store\Model\Store;
 use Magento\Store\Model\StoreManagerInterface;
 
 class ProductHelper
 {
+    public $productCollectionFactory;
+    public $groupCollection;
```

Comment:
Could you please make these 2 properties private ?

Reply:
I'd rather not. Shall we compromise and make them protected? Private leaves me with very few nice options to override these.

Reply:
(Right now we have a composer patch in place that makes all private properties and functions in your extension public so we can pluginize/override them - this isn't preferred)